lib/async_hooks.js

```diff
@@ -171,7 +171,7 @@ class AsyncResource {
 
   runInAsyncScope(fn, thisArg, ...args) {
     const asyncId = this[async_id_symbol];
-    emitBefore(asyncId, this[trigger_async_id_symbol]);
```

Comment:
I think the `emitAfter()` a few lines below needs also `this` added (can't add a comment there).

Reply:
After only pops the resource from the stack, so I don’t think it needs it, does it? Do you have a use-case in mind?